### PR TITLE
fix: remove unnecessary abis parameter from signer and .execute

### PR DESCRIPTION
**Pull request by PhilippeR26**
## Motivation and Resolution
Solve issue #1030 .
`Account.execute()` uses an optional `abis` parameter that is useless. Same for `Signer` interface.

- Signer optional `abis` parameter is deleted.
- `Account.execute()` accepts both signatures with and without `abis` parameter (to not break the current syntax).

## Usage related changes
All these syntaxes are valid :
```typescript
const res1 = await account0.execute(myCall,undefined,{maxFee:10n**17n});
const res2 = await account0.execute(myCall);
const res3 = await account0.execute(myCall,{maxFee:10n**17n});
```

## Development related changes
- Abis removed of Signer interface
- Account interface with execute method overloaded.
- Added a test of the new syntax.

**Nota** : Guides are automatically modified by CI during commit. I think that this has to be deactivated. Documentation content has to be handled only by the author.

## Checklist:

- [x] Performed a self-review of the code
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Linked the issues which this PR resolves
- [x] Documented the changes in code (API docs will be generated automatically)
- [x] Updated the tests
- [x] All tests are passing

### CODE REVIEW

File: __tests__/account.test.ts [hunk 1]
```diff
@@ -370,6 +370,19 @@ describe('deploy and test Wallet', () => {
...
+      {
+        contractAddress: erc20Address,
+        entrypoint: 'transfer',
+        calldata: [erc20.address, '10', '0'],
+      },
+      { maxFee: 6n * 10n ** 15n }
+    );
+
+    await provider.waitForTransaction(transaction_hash);
+  });
```
Comment (penovicp) [inline __tests__/account.test.ts:384]:
```suggestion
  test('execute with and without deprecated abis parameter', async () => {
    const transaction = {
      contractAddress: erc20Address,
      entrypoint: 'transfer',
      calldata: [erc20.address, '10', '0'],
    };
    const details = { maxFee: 0n };

    await expect(account.execute(transaction, details)).rejects.toThrow(/zero/);
    await expect(account.execute(transaction, undefined, details)).rejects.toThrow(/zero/);
  });
```
The suggestion makes the test explicitly about the parameter notation.

File: src/account/default.ts [hunk 2]
```diff
@@ -305,11 +305,34 @@ export class Account extends Provider implements AccountInterface {
...
+    transactionsDetail?: UniversalDetails
+  ): Promise<InvokeFunctionResponse>;
+  public async execute(
+    transactions: AllowArray<Call>,
+    abis?: Abi[],
+    transactionsDetail?: UniversalDetails
+  ): Promise<InvokeFunctionResponse>;
+  public async execute(
+    transactions: AllowArray<Call>,
+    arg2: Abi[] | UniversalDetails | undefined = undefined,
```
Comment (penovicp) [inline src/account/default.ts:318]:
```suggestion
    arg2?: Abi[] | UniversalDetails,
```

File: src/account/default.ts [hunk 3]
```diff
@@ -305,11 +305,34 @@ export class Account extends Provider implements AccountInterface {
...
+        break;
+      case typeof arg2 === 'object':
+        details = arg2;
+        break;
+      case typeof arg2 === 'undefined':
+        details = transactionsDetail;
+        break;
+      default:
+        throw Error(`wrong input parameters: ${arg2}, ${transactionsDetail}`);
+    }
```
Comment (penovicp) [inline src/account/default.ts:334]:
```suggestion
    const details = arg2 === undefined || Array.isArray(arg2) ? transactionsDetail : arg2;
```
This omits the explicit error but parameter validation is rarely used in the library and the code will fail anyway for faulty parameters.

**Review by penovicp — COMMENTED** [review]
Left some suggestions, in general lgtm

**Review by PhilippeR26 — COMMENTED** [review]
All done.

**Review by ivpavici — APPROVED** [review]

**Review by penovicp — APPROVED** [review]

**Pull request merged**